### ISSUE TITLE
[feature request] Split + name output files using designated Field Values 

### ISSUE THREAD

**Issue by ericjarvies** [mbloch/mapshaper#123]
`mapshaper sourceFile.shp -split -o format=geojson`
outputs the sourceFile name sans .shp extension + ascending number + new .json extension, e.g.-
_sourceFile-0.json
sourceFile-1.json
sourceFile-2.json_

`mapshaper sourceFile.shp -split -o format=geojson field=fid`
outputs file names using **Field Values** from designated **Field** in _sourceFile.shp_ **data table** (e.g.- fid) minus ascending number, e.g.-
_Record1fidFieldValue.json
Record2fidFieldValue.json
Record3fidFieldValue.json_

**Comment by mbloch** [comment]
What you are describing is very similar to the current behavior. In your second example above, you just need to move `field=fid` out of the `-o` command and into the `-split` command, like this:

`mapshaper sourceFile.shp -split field=fid -o format=geojson`
(as a shortcut you can write: `mapshaper sourceFile.shp -split fid -o format=geojson`)

When naming split-apart layers, mapshaper puts the layer name and a hyphen in front of the field value (see the sample output below). Do you prefer that the original layer name is not part of the split layer names?

Here's some actual program output:
```
$ mapshaper six_counties.shp -split FIPS -o format=geojson
Wrote six_counties-11001.json
Wrote six_counties-51013.json
Wrote six_counties-51059.json
Wrote six_counties-51510.json
Wrote six_counties-51600.json
Wrote six_counties-51683.json
```

**Comment by ericjarvies** [comment]
Ah.  Yes, I was placing it after the **-o** command.  Thank you for clarifying that.  Both of your aforementioned options work(ed) as described.

Regarding the original file/layer name, yes, I'd like to output without it.  My **fid** (or **PID** rather) field contains **Property ID** numbers that match the geometry, and I'd like to name the files using their **PID** numbers so they are easily referenced (manually or programmatically).

You've got some pretty cool programs btw.

Thank you.

**Comment by mbloch** [comment]
I can think of a few options for having only field values in the names of split-apart layers.

Option A would be for mapshaper to leave off the layer name from split-apart layers, like in your example above. (I usually like having the layer name in there, but on a couple of occasions I didn't want it).

Option B would be to use just the field value when the layer is unnamed, like this:
```
mapshaper six_counties.shp name='' -split FIPS -o format=geojson
```
Currently the output from the above command is the following, but I think I prefer just using the field value when the layer name is empty:
```
Wrote split-11001.json
Wrote split-51013.json
Wrote split-51059.json
Wrote split-51510.json
Wrote split-51600.json
Wrote split-51683.json
```

Option C would be to use string substitution for naming split layers, which would give the most flexibility, maybe like this:
```
mapshaper six_counties.shp -split FIPS name='{FIPS}' -o
```

I've thought about option C in the past, but I'm inclined to go with option B for simplicity's sake. Does that work for you?

**Comment by ericjarvies** [comment]
Option B is fine.  But perhaps I am not entirely understanding what you mean.  When I run;
`mapshaper sourceFile.shp name='' -split pid -o format=geojson`

I get;
```
Wrote split-019-082-029.json
Wrote split-019-082-037.json
Wrote split-019-082-045.json
Wrote split-019-082-053.json
```

So, it prefixes the file name with '**split-**.'  Are you saying changing the current behavior of Option B so that when `name=''` is used -it does **not** insert any prefix?

**Comment by mbloch** [comment]
Yes, that's what I'm proposing. I doubt this will adversely affect anyone, so I'll plan to include it in the next release, probably within a week.

**Comment by mbloch** [comment]
Commit bb66de6cdde66de8477ac0949ce  adopts option B above.

**Comment by mbloch** [comment]
Just published v0.3.20, which includes this change, and updated the wiki.

**Comment by ericjarvies** [comment]
Just tested v0.3.20 and 
`mapshaper states.shp name='' -split STATE -o format=geojson`
works great!
